### PR TITLE
HPCC-23400 WIP - Integrate ELK components into HPCC helm

### PR DESCRIPTION
**Pull request by rpastrana**
- Declare Elasticsearch, filebeat, and kibana as dependancy
- Provide localELK switch

Signed-off-by: Rodrigo Pastrana <Rodrigo.Pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
Deploys an ELK environment alongside HPCC
perform a "helm dependency update" command at the HPCC-Platform/helm/hpcc level  
The elk chart packages will be saved in ./charts
>Saving 4 charts
>Downloading elasticsearch from repo https://helm.elastic.co
>Downloading filebeat from repo https://helm.elastic.co
>Downloading logstash from repo https://helm.elastic.co
>Downloading kibana from repo https://helm.elastic.co

kubectl get pods
NAME                                READY   STATUS              RESTARTS   AGE
eclqueries-6bbf679465-d8gcp         1/1     Running             0          41s
eclservices-59d7fbbfbf-fcgmc        0/1     Running             0          41s
eclwatch-769546bdd4-qpfnl           0/1     Running             0          41s
**elasticsearch-master-0              0/1     Running             0          40s
elasticsearch-master-1              0/1     Init:0/1            0          40s
elasticsearch-master-2              0/1     Running             0          40s**
esdl-sandbox-6b4dbdcc67-dt7hr       0/1     Running             0          41s
hthor-57f854ff77-bbnt8              0/1     Running             0          42s
**mycluster-filebeat-s6ddd            0/1     Running             0          42s
mycluster-filebeat-vnwrk            0/1     Running             0          42s
mycluster-filebeat-xw2tp            0/1     Running             0          42s
mycluster-kibana-578f7dcb9c-nj6cz   0/1     Running             0          42s
mycluster-logstash-0                0/1     ContainerCreating   0          40s**
mydali-65d75c76c4-5zs8l             1/1     Running             0          42s
myeclccserver-66c55578ff-tdr54      1/1     Running             0          42s
roxie-agent-1-77969568bc-n8gg7      0/1     Running             0          41s
roxie-agent-2-d7f54c548-8d8hq       1/1     Running             0          40s
roxie-toposerver-57ffdbc86-z576t    1/1     Running             0          41s
roxie-workunit-6d9db87f69-9488q     1/1     Running             0          42s
sql2ecl-b7d6fb5f9-wfnvh             0/1     Running             0          41s
thor-eclagent-69c6968c64-4jgl9      1/1     Running             0          40s
thor-thoragent-74888fcb96-llvl4     1/1     Running             0          40s

**Comment by richardkchapman** [comment]
I suspect this should target 8.0

Also please fill in the checkboxes

File: helm/hpcc/Chart.yaml [hunk 1]
```diff
@@ -10,4 +10,37 @@ version: 0.1.0
...
+
+# Requires 'helm dependency update' command
+dependencies:
+# An Elastic Stack environment is managed by default
+# To disable: --set managedelasticstack=false
+# and speficy alternate log processor
+
+- name: elasticsearch # Optional managed logging back-end store
+  version: 7.6.1
+  #repository: file://./charts/elasticsearch # Use local customized chart due to a couple of issues in elastic's helm
```
Comment (richardkchapman) [inline helm/hpcc/Chart.yaml:23]: This line should probably be removed?

File: helm/hpcc/values.schema.json [hunk 2]
```diff
@@ -393,10 +407,77 @@
         "detail": {
           "type": "integer",
           "description": "Log output verbosity"
+        },
```
Comment (richardkchapman) [inline helm/hpcc/values.schema.json:410]: Indentation looks a bit odd here - tabs perhaps?

**Review by richardkchapman — CHANGES_REQUESTED** [review]
I know it's work in progress, but a couple of comments so far

File: helm/hpcc/values.schema.json (hunk 2 above)
Comment (rpastrana) [reply helm/hpcc/values.schema.json:410]: Fixed, and added the githook to avoid tabs

**Pull request closed without merging**